src/components/App.js

```diff
@@ -78,6 +83,25 @@ const App = ( { children }: Props ): Node => {
   useObservationUpdatesWhenFocused( );
   useTaxonCommonNames( );
 
+  const showStorageFullAlert = () => Alert.alert(
+    "Device Storage Full",
+    "iNaturalist may not be able to save your photos or may crash",
+    [{ text: t( "OK" ) }],
+    {
+      cancelable: true
+    }
+  );
```

Comment:
This gets redefined on every render. Either stabilize it with useCallback or define it inside the useEffect callback so it only gets defined with the effect runs.